### PR TITLE
Add stub file for xml/dom/xmlbuilder.py

### PR DESCRIPTION
**Pull request by AlexWaygood**
I made a stab at adding stubs for `xml/dom/xmlbuilder.py` (cpython source code [here](https://github.com/python/cpython/blob/bb3e0c240bc60fe08d332ff5955d54197f79751c/Lib/xml/dom/xmlbuilder.py)). A little tricky, since documentation is rather thin on the ground, but perhaps an improvement on what's there at the moment?

### CODE REVIEW

**Comment by AlexWaygood** [comment]
The errors in the failing tests are:
- "error: xml.dom.xmlbuilder.ExpatBuilder is not present at runtime"
- "error: xml.dom.xmlbuilder.ExpatBuilderNS is not present at runtime"

Have I done something wrong there? The `CONTRIBUTING.md` file said: "Imports in stubs are considered private (not part of the exported API)."

Also, I included several methods that always raised errors, annotating them as returning `NoReturn` -- is that preferred, or should I have left them out?

**Comment by Akuli** [comment]
The CI failure looks like a bug to me. It is most likely a bug in mypy, as stubtest uses mypy internally. It might be https://github.com/python/mypy/issues/10661

**Comment by hauntsaninja** [comment]
Is the failure a bug? I genuinely don't see these two at runtime:
```
λ python
Python 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import xml.dom.xmlbuilder
>>> xml.dom.xmlbuilder.ExpatBuilder
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'xml.dom.xmlbuilder' has no attribute 'ExpatBuilder'
>>> xml.dom.xmlbuilder.ExpatBuilderNS
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'xml.dom.xmlbuilder' has no attribute 'ExpatBuilderNS'
```

**Comment by hauntsaninja** [comment]
Okay, confirmed mypy bug. Here's the fix: https://github.com/python/mypy/pull/11411

For now, I recommend adding it to stubtest's allowlist here: https://github.com/python/typeshed/blob/master/tests/stubtest_allowlists/py3_common.txt

If you then also want to run isort against this PR, CI will be green

**Comment by AlexWaygood** [comment]
> Okay, confirmed mypy bug. Here's the fix: [python/mypy#11411](https://github.com/python/mypy/pull/11411)
> 
> For now, I recommend adding it to stubtest's allowlist here: https://github.com/python/typeshed/blob/master/tests/stubtest_allowlists/py3_common.txt
> 
> If you then also want to run isort against this PR, CI will be green

🥳 CI is now green -- thanks!

File: stdlib/xml/dom/xmlbuilder.pyi [hunk 1]
```diff
@@ -1,6 +1,142 @@
...
+    def _get_errorHandler(self) -> _DOMBuilderErrorHandlerType: ...
+    def _set_errorHandler(self, errorHandler: _DOMBuilderErrorHandlerType) -> None: ...
+    def _get_filter(self) -> DOMBuilderFilter | None: ...  # a guess, but seems likely
+    def _set_filter(self, filter: DOMBuilderFilter | None) -> None: ...  # a guess, but seems likely
+    def setFeature(self, name: str, state: int) -> None: ...
+    def supportsFeature(self, name: str) -> bool: ...
+    def canSetFeature(self, name: str, state: int) -> bool: ...
+    _settings: dict[tuple[str, int], list[tuple[str, int]]]
+    # getFeature could return any attribute from an instance of `Options`
+    def getFeature(self, name: str) -> int | bool | None | Any: ...
```
Comment (Akuli) [inline stdlib/xml/dom/xmlbuilder.pyi:70]: This should probably just return `Any`, unless we add a big pile of overloads.

File: stdlib/xml/dom/xmlbuilder.pyi [hunk 2]
```diff
@@ -1,6 +1,142 @@
...
+    ACTION_INSERT_BEFORE: Literal[4]
+    _legal_actions: tuple[Literal[1], Literal[2], Literal[3], Literal[4]]
+    def __init__(self) -> None: ...
+    _options: Options
+    def _get_entityResolver(self) -> DOMEntityResolver | None: ...  # a guess, but seems likely
+    def _set_entityResolver(self, entityResolver: DOMEntityResolver | None) -> None: ...  # a guess, but seems likely
+    def _get_errorHandler(self) -> _DOMBuilderErrorHandlerType: ...
+    def _set_errorHandler(self, errorHandler: _DOMBuilderErrorHandlerType) -> None: ...
+    def _get_filter(self) -> DOMBuilderFilter | None: ...  # a guess, but seems likely
+    def _set_filter(self, filter: DOMBuilderFilter | None) -> None: ...  # a guess, but seems likely
```
Comment (Akuli) [inline stdlib/xml/dom/xmlbuilder.pyi:64]: Is there any reason to include all these `_non_public` methods?

File: tests/stubtest_allowlists/py3_common.txt [hunk 3]
```diff
@@ -217,6 +217,8 @@ webbrowser.UnixBrowser.remote_action  # always overridden in inheriting class
 webbrowser.UnixBrowser.remote_action_newtab  # always overridden in inheriting class
 webbrowser.UnixBrowser.remote_action_newwin  # always overridden in inheriting class
 wsgiref.types  # Doesn't exist, see comments in file
+xml.dom.xmlbuilder.ExpatBuilder
+xml.dom.xmlbuilder.ExpatBuilderNS
```
Comment (Akuli) [inline tests/stubtest_allowlists/py3_common.txt:221]: Why are these whitelisted?

**Review by Akuli — COMMENTED** [review]
It seems that everyone somehow missed this :)

File: tests/stubtest_allowlists/py3_common.txt (hunk 3 above)
Comment (Akuli) [reply tests/stubtest_allowlists/py3_common.txt:221]: I now found the discussion above. Can you add a comment that mentions https://github.com/python/mypy/pull/11411 to these lines?
```

Comment (AlexWaygood) [reply tests/stubtest_allowlists/py3_common.txt:221]: Yes!

> It seems that everyone somehow forgot about this :)

The PR rises from the dead!

**Comment by AlexWaygood** [comment]
> It seems that everyone somehow forgot about this :)

The PR rises from the dead!

**Comment by Akuli** [comment]
You need to merge the latest `master` into the branch of this PR.

**Comment by AlexWaygood** [comment]
> You need to merge the latest `master` into the branch of this PR.

Oh, I think I understand how this happened.

**Comment by Akuli** [comment]
This PR conflicted with #6379, which is not surprising because #6379 changed many lines.

File: stdlib/xml/dom/xmlbuilder.pyi [hunk 4]
```diff
@@ -1,6 +1,136 @@
...
+    def _get_stringData(self) -> _DOMInputSourceStringDataType: ...
+    def _set_stringData(self, data: _DOMInputSourceStringDataType) -> None: ...
+    def _get_encoding(self) -> _DOMInputSourceEncodingType: ...
+    def _set_encoding(self, encoding: _DOMInputSourceEncodingType) -> None: ...
+    def _get_publicId(self) -> str | None: ...
+    def _set_publicId(self, publicId: str | None) -> None: ...
+    def _get_systemId(self) -> str | None: ...
+    def _set_systemId(self, systemId: str | None) -> None: ...
+    def _get_baseURI(self) -> str | None: ...
+    def _set_baseURI(self, uri: str | None) -> None: ...
```
Comment (Akuli) [inline stdlib/xml/dom/xmlbuilder.pyi:103]: There's still quite a lot of `_non_public` methods, and also a few attributes in `DOMBuilder`.

File: stdlib/xml/dom/xmlbuilder.pyi [hunk 5]
```diff
@@ -1,6 +1,109 @@
...
+    FILTER_INTERRUPT: Literal[4]
+    whatToShow: int
+    # The argtypes for acceptNode and startContainer appear to be irrelevant.
+    def acceptNode(self, element: object) -> Literal[1]: ...
+    def startContainer(self, element: object) -> Literal[1]: ...
+
+class DocumentLS:
+    async_: bool
+    # The only requirements on the `flag` argument
+    # are that it can safely be passed to `bool()`.
```
Comment (Akuli) [inline stdlib/xml/dom/xmlbuilder.pyi:94]: This comment no longer applies, because the method that had a `flag` argument was deleted.

Comment (AlexWaygood) [reply stdlib/xml/dom/xmlbuilder.pyi:94]: lol

Comment (AlexWaygood) [reply stdlib/xml/dom/xmlbuilder.pyi]: thanks!

**Review by Akuli — APPROVED** [review]
Much better than what was there before.

**Comment by AlexWaygood** [comment]
> Much better than what was there before.

Thank you for the patient reviews :)

**Comment by AlexWaygood** [comment]
🙌🙌

**Pull request merged**